### DISCUSSION TITLE
My Haxophone

### DISCUSSION THREAD

**Discussion by raychmond** [cardonabits/haxo-hw#15] [Show and tell]
![CB1792E9-4B44-48A1-B195-D4C6537F700F](https://user-images.githubusercontent.com/60025294/229670302-5f3a41c0-81cd-427e-a76b-fd50c0293b69.jpeg)
![D88C90CF-CB1D-451F-A98B-5AC2AF02011F](https://user-images.githubusercontent.com/60025294/229670306-26ccac89-662f-4d8f-a3d2-fa1febf77b2e.jpeg)

**Comment by raychmond** [comment]
Thank you so much @jcard0na for your endless support! It’s now working through the raspberry pi’s audio out. The sound is already beautiful. I believe it will be great if it uses the on board amp.

**Reply by bonjourrichard** [reply]
Your haxophone looks great!  It's an inspiration for me to start such a project.  I've asked my sons to help me build it for Father's Day.   @raychmond, can you share a BOM price and a list of component vendors that you used to develop a single instrument?  Many thanks!

**Reply by bonjourrichard** [reply]
@raychmond, I see the link to your BOM in your other comment.  Thank you!

**Reply by raychmond** [reply]
@bonjourrichard I ordered most parts from the BOM link in this post. Depending on your location, you may choose different vendors. I bought most of them through Newark. The PCB is from JCLPCB. I ordered the food grade silicon tubing from a UK company, but I haven't received yet. I'm using a non-food grade tubing with OD:5mm. It's working, but I believe OD:6mm will be better.
Please let me know if you have any questions  while you're building your own haxophone. BTW, it's awesome! I play it everyday.

**Reply by bonjourrichard** [reply]
@raychmond , Thank you for the additional information.  Any bits of knowledge will give my sons and me more confidence to start.

Once we achieve success (Target Date: June 18, 2023), then, like you, I will also post photos and notes. 

@jcard0na Javier, once again, thank you for engineering this project!

**Reply by jcard0na** [reply]
> Once we achieve success (Target Date: June 18, 2023), then, like you, I will also post photos and notes.

Marking that date on my calendar.  Don't hesitate to ask questions or report any [issues](https://github.com/jcard0na/haxo-hw/issues) you encounter while working on the project.

**Comment by jcard0na** [comment]
Wow, @raychmond!
This is a beautiful instrument you've built!  Congratulations!

**Reply by raychmond** [reply]
@jcard0na 
Thank you so much for the beautiful design and your help on building my own haxo. I'm now learning how to play it. I know how to play flute, but not saxophone. It now becomes much easier, at least I won't make terrible noise as I did on a saxophone. My daughter is a saxophone player, she's so surprised that it makes such awesome sound and really playable.

Thanks!

**Reply by jcard0na** [reply]
You are welcome!

By the way, this is a little tool I wrote to allow editing the note mapping.   Given that you are coming from the flute, you might want to customize your fingering: https://haxo-notemap.nn.r.appspot.com/

Cheers,

**Reply by jcard0na** [reply]
@raychmond now that you've completed the full assembly successfully, would you mind taking a second look at [the assembly instructions](https://github.com/jcard0na/haxo-hw/blob/main/docs/assembly.md) and let me know if I missed anything, or if there is any step that requires additional clarification/pictures?  Feel free to do a [pull request](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) with your own suggestions and/or pictures to those instructions.  This way you'll show up in the very selective group of [project contributors](https://github.com/jcard0na/haxo-hw/graphs/contributors) 😉  
Also, is [the BOM](https://github.com/jcard0na/haxo-hw#bill-of-materials) still valid?  Where you able to find all those parts or you had to find replacements?  Thank you on behalf of those who might follow your footsteps...

Cheers,

**Reply by raychmond** [reply]
Hi Javier @jcard0na 
The overall experience was great! I had no problem assembling the hardware.
I will let you know some my thoughts and experience on the software installation. I also have a couple of ideas. I will initiate the pull request later.

Thank you!

**Reply by raychmond** [reply]
As I said, the overall experience assembling the hardware was great. I followed every step in the assembly instructions and bought the parts from this BOM [link](https://octopart.com/bom-tool/1uKwuwiE). The L shape brackets are very special. We need smaller hole thread to act as the nut. So don't buy a random L shape brackets.
The only issue I ran into was when assembling the left and right panels to the main board, i.e. step 11. My right-angle 4 pin connectors were too tight in the main board, so I had to use force to bend them to connect with the side panels. My suggestion is not to rush to solder the right-angle connectors in step 3. You can complete the side panels first, connect it with the right-angle connectors and solder them onto the main board. Or at least check step 11 first and make sure you have enough space between the right-angle pins and the main board.
Other lesson I've learned is to take the "General Advice" seriously. Raspberry Pi is really sensitive to the leakage path. I made another device using the teensy 4.1. I basically soldered 20+ same type of keyswitchs using the same type of soldering. I didn't run into any problem. But I have 6 keys with the leakage path and didn't pass the test this time. I guess the only difference is the main processor.
I will share my experience when installing the software next time. 
Thanks!

**Reply by jcard0na** [reply]
Thanks!  I'll add those suggestions to the assembly instructions.  Thanks!

**Comment by bonjourrichard** [comment]
@jcard0na, your haxophone looks wonderful!  I can't wait to build it.  Even more, I can't wait to play it.   I travel for work, and this appears to be an instrument that I can put into my computer bag. Just wow!  Thank you!

**Reply by jcard0na** [reply]
It definitely fits in my computer bag, and sometimes I just use my laptop to power up the haxophone, skipping the need for a power source.  I really hope it works for you!

**Reply by raychmond** [reply]
![B23AC86D-F753-4D9C-A7F0-9B502EBA43DE](https://user-images.githubusercontent.com/60025294/232351067-6af03894-bb37-4fd2-a524-43f8932afb67.jpeg)
I use a [UPS-Lite](https://www.aliexpress.com/item/32961105269.html?spm=a2g0o.order_list.order_list_main.101.19701802HB2g5t) on the back of RPi 4B. It works very well. I believe it can power my 4B at least one hour. If I change 4B to zero later, I think it can power even longer. This is a great portable solution if you want to use Haxophone while traveling.

**Comment by jcard0na** [comment]
Hi @raychmond,

I'm preparing a short video for the [haxophone pre-order campaign](https://www.crowdsupply.com/cardona-bits/haxophone).  This is intended for people who want a pre-assembled instrument.  In that video, I want to show how easy it is to customize your instrument.  As an example, I was going to show how you can change the key colors very quickly by replacing keycaps.  I was going to use my horn, but I find that your colors are nicer than mine!  Would you be willing to share a few seconds of video raplacing a keycap on your instrument and let me use it in the campaign?  You can do that with your phone, [these are the guidelines for how to record](https://www.crowdsupply.com/guide/project-video#shooting-tips).  

If you want to achieve worldwide fame as a haxophonist (we aren't that many out there, you know... 😉 ) you could also share footage of you rehearsing your favourite exercise.  Feel free to ignore, if you don't find this fun or you are too busy!

Cheers,

P.S.  Your UPS-Lite hack is a great tip.  I've added a mention to it to issue #4.

**Reply by raychmond** [reply]
Thanks @jcard0na 
I'm definitely happy to contribute. I will make the video changing the keycaps and share here. In terms of the footage of rehearsing, I may need a few more days to practice :-), but yeah, I'm certainly willing to share.

**Comment by mikenturner** [comment]
Hi Raymond, 
Javier suggested I contact you in case you have a PCB spare from your last order?
Thanks
Mike

**Reply by raychmond** [reply]
I'm sorry Mike. I have started soldering parts on the only left PCB board. I'm building the second one as a gift to my daughter. Ordering PCB, parts, and building the hardware are relatively straight forward. I'm sure you can find someone else who are interested as well to share the cost. The software part is a little bit tricky. I will write my experience soon. But I'm just too busy at work recently. 
Thanks!
